libs/tests/core/test_toolkit.py

```diff
@@ -0,0 +1,344 @@
+    """Test the combined toolkit discovery method."""
+
+    @patch("arcade_core.toolkit.Toolkit.find_arcade_toolkits_from_entrypoints")
+    @patch("arcade_core.toolkit.Toolkit.find_arcade_toolkits_from_prefix")
+    def test_find_all_no_duplicates(self, mock_find_prefix, mock_find_ep):
+        """Test that find_all returns combined results without duplicates."""
+        # Create mock toolkits
+        toolkit1 = Toolkit(
+            name="toolkit1",
+            package_name="toolkit1",
+            version="1.0.0",
+            description="Toolkit 1",
+        )
+        toolkit2 = Toolkit(
+            name="toolkit2",
+            package_name="toolkit2",
+            version="1.0.0",
+            description="Toolkit 2",
+        )
+        # Mock the discovery methods
+        mock_find_ep.return_value = [toolkit1]
+        mock_find_prefix.return_value = [toolkit2]
+
+        toolkits = Toolkit.find_all_arcade_toolkits()
+
+        assert len(toolkits) == 2
+        assert any(t.name == "toolkit1" for t in toolkits)
+        assert any(t.name == "toolkit2" for t in toolkits)
+
+    @patch("arcade_core.toolkit.Toolkit.find_arcade_toolkits_from_entrypoints")
+    @patch("arcade_core.toolkit.Toolkit.find_arcade_toolkits_from_prefix")
+    def test_find_all_with_duplicates_prefers_entrypoint(self, mock_find_prefix, mock_find_ep):
+        """Test that entry point toolkits are preferred over prefix-based ones."""
+        toolkit_ep = Toolkit(
+            name="test",
+            package_name="arcade_test",  # Same package name
+            version="2.0.0",
+            description="Entry point version",
+        )
+
+        toolkit_prefix = Toolkit(
+            name="test",
+            package_name="arcade_test",  # Same package name
+            version="1.0.0",
+            description="Prefix version",
+        )
+        # Mock the discovery methods
+        mock_find_ep.return_value = [toolkit_ep]
+        mock_find_prefix.return_value = [toolkit_prefix]
+
+        toolkits = Toolkit.find_all_arcade_toolkits()
+
+        assert len(toolkits) == 1
+        assert toolkits[0].version == "2.0.0"
+        assert toolkits[0].description == "Entry point version"
+
+    @patch("arcade_core.toolkit.Toolkit.find_arcade_toolkits_from_entrypoints")
+    @patch("arcade_core.toolkit.Toolkit.find_arcade_toolkits_from_prefix")
+    def test_find_all_empty(self, mock_find_prefix, mock_find_ep):
+        """Test when no toolkits are found."""
+        mock_find_ep.return_value = []
+        mock_find_prefix.return_value = []
+
+        toolkits = Toolkit.find_all_arcade_toolkits()
+        assert toolkits == []
+
+
+class TestEntryPointCompatibility:
+    """Test compatibility scenarios for entry point discovery."""
+
+    @patch("arcade_core.toolkit.importlib.metadata.entry_points")
+    @patch("arcade_core.toolkit.Toolkit.from_entrypoint")
+    def test_duplicate_toolkit_names_in_entrypoints(self, mock_from_ep, mock_entry_points):
```

Comment:
question for the reviewer(s):

The behavior is that we now allow for more than 1 toolkit to have the same name as long as their package names are different. Do we want this behavior?
This means we could have something like:
```
`Github.ListStargazers`  # from the official arcade-github package
`Github.ListIssues` # from a community package named joe-shmoe-github
```